blog/modules/ROOT/pages/sort-antora-components.adoc

```diff
@@ -0,0 +1,196 @@
+= Awesome Antora: Sort and hide components and versions
+David Jencks <djencks@apache.org>
+:page-revdate: 2020-9-28 14:25
+:page-author: David Jencks
+:description: You would like custom sorting of Antora components and versions, or to hide components.
+:page-tags: Antora, Sorting Components
+:page-image: sort.jpg
+:page-color-primary: #8BB3B2
+:page-color-secondary: #C68B45
+:ms-sorter-url: https://gitlab.com/opendevise/oss/antora-site-generator-ms/-/blob/v2.3.x/lib/custom-sort-components.js
+
+A frequent request is to sort the components and their versions in the default Antora UI component explorer in a customized order.
+Fortunately, this is quite easy to do using a couple of files in a supplemental UI or in a customized UI.
+I'd like to thank Bob Roro and Ewan Edwards for pushing me into figuring out how this works and a lot of hints and answers and Dan Allen for pointing to an link:{ms-sorter-url}[existing implementation] (for a completely custom UI).
+
+== Where do we sort?
+
+Here's the default UI's partial that sets up the component explorer contents:
+
+.nav-explore.hbs (original)
+[source,handlebars]
+----
+include::example$sort-components/nav-explore-default.hbs[]
+----
+<1> The crucial line!
+
+Here, `{{#each site.components}}` iterates over the object `site.components` in the default order.
+All we have to do is use a javascript helper to change the order, as in `{{#each (mySorter site.components)}}`.
+For this the javascript helper signature would be `module.exports = (components) => {...}`.
+Note that `#each` works on either arrays or objects, so, although `site.components` is an object, we can return an array of components in the desired order.
+
+Now lets consider plausible sources of information for the sorter.
+First of all, we need the collection of components.
+This is actually available in two ways, as `site.components` or by calling `getComponents` on the content catalog model as `site.contentCatalog`.
+We could just pass the entire `site` model to our helper, but lets just use `site.components` like the default.
```

Comment:
I think using "just" should be avoided even though in this case it might OK, what do you think?

```suggestion
We could pass the entire `site` model to our helper, but lets use `site.components` like the default.
```